src/execution_plan/execution_plan.c

```diff
@@ -510,56 +517,88 @@ static void _ExecutionPlan_FreeInternals(ExecutionPlan *plan) {
 			QueryGraph_Free(plan->connected_components[i]);
 		}
 		array_free(plan->connected_components);
+		plan->connected_components = NULL;
 	}
 
 	if(plan->query_graph) {
 		QueryGraph_Free(plan->query_graph);
+		plan->query_graph = NULL;
 	}
 	if(plan->record_map != NULL) {
 		raxFree(plan->record_map);
+		plan->record_map = NULL;
 	}
 	if(plan->record_pool != NULL) {
 		ObjectPool_Free(plan->record_pool);
+		plan->record_pool = NULL;
 	}
 	if(plan->ast_segment != NULL) {
 		AST_Free(plan->ast_segment);
+		plan->ast_segment = NULL;
 	}
 	rm_free(plan);
 }
 
 // Free an op tree and its associated ExecutionPlan segments.
-static ExecutionPlan *_ExecutionPlan_FreeOpTree(OpBase *op) {
-	if(op == NULL) return NULL;
-	ExecutionPlan *child_plan = NULL;
-	ExecutionPlan *prev_child_plan = NULL;
-	// Store a reference to the current plan.
-	ExecutionPlan *current_plan = (ExecutionPlan *)op->plan;
+static void _ExecutionPlan_FreeOpTree(OpBase *op) {
+	if(op == NULL) {
+		return;
+	}
+
 	for(uint i = 0; i < op->childCount; i ++) {
-		child_plan = _ExecutionPlan_FreeOpTree(op->children[i]);
-		// In most cases all children will share the same plan, but if they don't
-		// (for an operation like UNION) then free the now-obsolete previous child plan.
-		if(prev_child_plan != child_plan && prev_child_plan != current_plan) {
-			_ExecutionPlan_FreeInternals(prev_child_plan);
-			prev_child_plan = child_plan;
-		}
+		_ExecutionPlan_FreeOpTree(op->children[i]);
 	}
 
 	// Free this op.
 	OpBase_Free(op);
+}
 
-	// Free each ExecutionPlan segment once all ops associated with it have been freed.
-	if(current_plan != child_plan) _ExecutionPlan_FreeInternals(child_plan);
+static void _ExecutionPlan_AggregatePlansFromOps
+(
+	OpBase *opBase,     // operation to aggregate plans from
+	rax *plans          // plans map to insert to
+) {
+	ASSERT(plans != NULL);
 
-	return current_plan;
+	// add the plan if it doesn't exist already
+	ASSERT(sizeof(opBase->plan) == sizeof(uint64_t));
+	const uint64_t plan_ptr_be = _htobe64((uint64_t)opBase->plan);
```

Comment:
Why (if at all) is this necessary?

Reply:
Do you mean why the _htobe64 is necessary?